### PR TITLE
[CHORE] Update react-navigation to v5

### PR DESCRIPTION
**Pull request by diegolmello**
@RocketChat/ReactNative

### CODE REVIEW

File: app/views/RoomsListView/ServerDropdown.js [hunk 1]
```diff
@@ -225,12 +227,14 @@ class ServerDropdown extends Component {
 
 	render() {
 		const { servers } = this.state;
-		const { theme } = this.props;
+		const { theme, isMasterDetail } = this.props;
 		const maxRows = 4;
 		const initialTop = 41 + (Math.min(servers.length, maxRows) * ROW_HEIGHT);
+		const statusBarHeight = isIOS ? 20 : 0;
+		const heightDestination = isMasterDetail ? headerHeight + statusBarHeight : 0;
```
Comment (djorkaeffalexandre) [inline app/views/RoomsListView/ServerDropdown.js:234]: Screen Shot 2020-06-05 at 13 04 23

File: app/views/RoomView/index.js [hunk 2]
```diff
@@ -357,15 +280,87 @@ class RoomView extends React.Component {
...
 	}
 
-	// eslint-disable-next-line react/sort-comp
-	goRoomActionsView = () => {
-		const { room, member } = this.state;
-		const { navigation } = this.props;
-		navigation.navigate('RoomActionsView', {
-			rid: this.rid, t: this.t, room, member
+	setHeader = () => {
+		const { room, unreadsCount, roomUserId: stateRoomUserId } = this.state;
```
Comment (djorkaeffalexandre) [inline app/views/RoomView/index.js:284]: `unreadsCount` always is null here

File: app/views/RoomsListView/index.js [hunk 3]
```diff
@@ -190,29 +144,20 @@ class RoomsListView extends React.Component {
...
 		if (isTablet) {
 			EventEmitter.addEventListener(KEY_COMMAND, this.handleCommands);
 		}
 		Dimensions.addEventListener('change', this.onDimensionsChange);
-		this.willFocusListener = navigation.addListener('willFocus', () => {
-			// Check if there were changes while not focused (it's set on sCU)
-			if (this.shouldUpdate) {
-				this.forceUpdate();
-				this.shouldUpdate = false;
-			}
```
Comment (djorkaeffalexandre) [inline app/views/RoomsListView/index.js:213]: Do we need this yet?

File: app/views/SearchMessagesView/index.js [hunk 4]
```diff
@@ -18,17 +17,25 @@ import StatusBar from '../../containers/StatusBar';
...
-		...themedHeader(screenProps.theme)
-	})
+	static navigationOptions = ({ navigation, route }) => {
+		const options = {
+			title: I18n.t('Search')
+		};
+		const showCloseModal = route.params?.showCloseModal;
+		if (showCloseModal) {
+			options.headerLeft = () => <CloseModalButton navigation={navigation} />;
+		}
```
[the remark this answers is not in the document]
Comment (diegolmello) [reply app/views/SearchMessagesView/index.js:32]: Cmd+F key command

File: app/containers/Header/index.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { SafeAreaView } from 'react-native-safe-area-context';
+import { View, StyleSheet } from 'react-native';
+import { themes } from '../../constants/colors';
+import { themedHeader } from '../../utils/navigation';
+import { isIOS } from '../../utils/deviceInfo';
+
+export const headerHeight = isIOS ? 44 : 56;
```
[the remark this answers is not in the document]
Comment (diegolmello) [reply app/containers/Header/index.js:9]: It's always returning 0.
I picked these values from https://github.com/react-navigation/react-navigation/blob/master/packages/stack/src/views/Header/HeaderSegment.tsx#L69

File: app/stacks/MasterDetailStack/index.js [hunk 6]
```diff
@@ -0,0 +1,353 @@
...
+			/>
+		</InsideStack.Navigator>
+	);
+});
+
+const CustomInsideStack = React.memo(({ navigation, route }) => (
+	<>
+		<InsideStackModal navigation={navigation} />
+		<NotificationBadge navigation={navigation} route={route} />
+		<Toast />
```
[the remark this answers is not in the document]
Comment (diegolmello) [reply app/stacks/MasterDetailStack/index.js:345]: I think it's correct.
It's an alert that may be fired from modals as well.
It's similar to bottom sheet or in-app notifications, that are aligned to the center of the screen.

File: app/views/CreateDiscussionView/index.js [hunk 7]
```diff
@@ -100,6 +81,23 @@ class CreateChannelView extends React.Component {
...
+			headerRight: (
+				this.valid()
+					? () => (
+						<CustomHeaderButtons>
+							<Item title={I18n.t('Create')} onPress={this.submit} testID='create-discussion-submit' />
+						</CustomHeaderButtons>
+					)
+					: null
+			),
+			headerLeft: () => <CloseModalButton navigation={navigation} />
```
[the remark this answers is not in the document]
Comment (diegolmello) [reply app/views/CreateDiscussionView/index.js:97]: I fixed the duplicated close button, but I kept this flow.
I don't see why we should show it as another modal for discussions, but we show as stack for channels or grouped DMs.
WDYT?

[the remark this answers is not in the document]
Comment (djorkaeffalexandre) [reply app/views/CreateDiscussionView/index.js:97]: Yeah! I liked this flow at the NewMessage screen! But I think this should be a modal at the MessageBox actions and from MessageActions.

File: app/share.js [hunk 8]
```diff
@@ -14,61 +14,114 @@ import {
...
+	const { theme } = useContext(ThemeContext);
+
+	const screenOptions = {
+		...defaultHeader,
+		...themedHeader(theme)
+	};
+	screenOptions.headerStyle = {
+		...screenOptions.headerStyle,
+		// TODO: fix on multiple files PR :)
+		height: 57
```
Comment (djorkaeffalexandre) [inline app/share.js:47]: I think shareListView should have this same value at Header
WhatsApp Image 2020-06-12 at 17 00 33
Look below the header, a weird margin.

Comment (diegolmello) [reply app/share.js:47]: It's going to be fixed on multiple files PR

**Pull request merged**